src/core/ddsi/src/ddsi_typewrap.c

```diff
@@ -1141,7 +1188,17 @@ static dds_return_t add_complete_typeobj (struct ddsi_domaingv *gv, struct xt_ty
       {
         xt->_u.union_type.members.seq[n].id = cto->_u.union_type.member_seq._buffer[n].common.member_id;
         xt->_u.union_type.members.seq[n].flags = cto->_u.union_type.member_seq._buffer[n].common.member_flags;
-        ddsi_type_register_dep (gv, &xt->id, &xt->_u.union_type.members.seq[n].type, &cto->_u.union_type.member_seq._buffer[n].common.type_id);
+        if ((ret = ddsi_type_register_dep (gv, &xt->id, &xt->_u.union_type.members.seq[n].type, &cto->_u.union_type.member_seq._buffer[n].common.type_id)) != DDS_RETCODE_OK)
+        {
+          for (uint32_t m = 0; m < n - 1; m++)
+          {
+            ddsi_type_unref_locked (gv, xt->_u.union_type.members.seq[m].type);
+            ddsrt_free (xt->_u.union_type.members.seq[m].label_seq._buffer);
```

Comment:
Shouldn't there be a `xt_applied_member_annotations_fini` here, too, like in the structure case?

Reply:
Yes, good catch, I've fixed this in [f65235c](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1557/commits/f65235ca3c8c75b06a7790cadab765302ca6400a)